toContain causes invalid output with array of objects

E.g.
```ts
const array = [];
Expect(array).toContain({});
```

Outputs:
```
100% complete
Warning: Failed prop type: Invalid prop `children` supplied to `Box`, expected a ReactNode.
    in Box (created by FailureDetail)
    in FailureDetail (created by TapBarkResults)
    in div (created by Box)
    in Box (created by TapBarkResults)
    in div (created by Box)
    in Box (created by TapBarkResults)
    in div (created by Static)
    in Static (created by TapBarkResults)
    in TapBarkResults (created by TapBarkRoot)
    in TapBarkRoot
    in App
The above error occurred in the <div> component:
    in div (created by Box)
    in Box (created by FailureDetail)
    in FailureDetail (created by TapBarkResults)
    in div (created by Box)
    in Box (created by TapBarkResults)
    in div (created by Box)
    in Box (created by TapBarkResults)
    in div (created by Static)
    in Static (created by TapBarkResults)
    in TapBarkResults (created by TapBarkRoot)
    in TapBarkRoot
    in App

React will try to recreate this component tree from scratch using the error boundary you provided, App.
Warning: App: Error boundaries should implement getDerivedStateFromError(). In that method, return a state update to display an error message or fallback UI.
(node:30660) UnhandledPromiseRejectionWarning: Error: Objects are not valid as a React child (found: object with keys {}). If you meant to render a collection of children, use an array instead.
    in div (created by Box)
    in Box (created by FailureDetail)
    in FailureDetail (created by TapBarkResults)
    in div (created by Box)
    in Box (created by TapBarkResults)
    in div (created by Box)
    in Box (created by TapBarkResults)
    in div (created by Static)
    in Static (created by TapBarkResults)
    in TapBarkResults (created by TapBarkRoot)
    in TapBarkRoot
    in App
    at throwOnInvalidObjectType (C:\code\_personal\uwhtraining\node_modules\react-reconciler\cjs\react-reconciler.development.js:4524:15)
    at reconcileChildFibers (C:\code\_personal\uwhtraining\node_modules\react-reconciler\cjs\react-reconciler.development.js:5382:7)
    at reconcileChildren (C:\code\_personal\uwhtraining\node_modules\react-reconciler\cjs\react-reconciler.development.js:7865:28)
    at updateHostComponent (C:\code\_personal\uwhtraining\node_modules\react-reconciler\cjs\react-reconciler.development.js:8387:3)
    at beginWork$1 (C:\code\_personal\uwhtraining\node_modules\react-reconciler\cjs\react-reconciler.development.js:9969:14)
    at Object.invokeGuardedCallbackImpl (C:\code\_personal\uwhtraining\node_modules\react-reconciler\cjs\react-reconciler.development.js:11563:10)
    at invokeGuardedCallback (C:\code\_personal\uwhtraining\node_modules\react-reconciler\cjs\react-reconciler.development.js:11740:31)
    at beginWork$$1 (C:\code\_personal\uwhtraining\node_modules\react-reconciler\cjs\react-reconciler.development.js:15778:7)
    at performUnitOfWork (C:\code\_personal\uwhtraining\node_modules\react-reconciler\cjs\react-reconciler.development.js:14696:12)
    at workLoopSync (C:\code\_personal\uwhtraining\node_modules\react-reconciler\cjs\react-reconciler.development.js:14669:22)
    at performSyncWorkOnRoot (C:\code\_personal\uwhtraining\node_modules\react-reconciler\cjs\react-reconciler.development.js:14265:11)
    at C:\code\_personal\uwhtraining\node_modules\react-reconciler\cjs\react-reconciler.development.js:1931:24
    at unstable_runWithPriority (C:\code\_personal\uwhtraining\node_modules\scheduler\cjs\scheduler.development.js:697:12)
    at runWithPriority (C:\code\_personal\uwhtraining\node_modules\react-reconciler\cjs\react-reconciler.development.js:1881:10)
    at flushSyncCallbackQueueImpl (C:\code\_personal\uwhtraining\node_modules\react-reconciler\cjs\react-reconciler.development.js:1926:7)
    at flushSyncCallbackQueue (C:\code\_personal\uwhtraining\node_modules\react-reconciler\cjs\react-reconciler.development.js:1914:3)
(node:30660) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 3)
(node:30660) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
npm ERR! Test failed.  See above for more details.
```

--- a/packages/tap-bark/package.json
+++ b/packages/tap-bark/package.json
@@ -39,11 +39,16 @@
         "through2": "^3.0.1"
     },
     "devDependencies": {
+        "@types/enzyme": "^3.10.7",
+        "@types/enzyme-adapter-react-16": "^1.0.6",
         "@types/react": "^16.9.17",
         "@types/through2": "^2.0.34",
+        "enzyme": "^3.11.0",
+        "enzyme-adapter-react-16": "^1.15.5",
         "install-self": "^1.0.0-rc.1",
         "nyc": "^14.1.1",
         "prettier": "^1.10.2",
+        "react-dom": "^16.13.1",
         "rimraf": "^3.0.0",
         "tslint": "^5.11.0",
         "tslint-plugin-prettier": "^1.3.0",

--- a/packages/tap-bark/src/components/failure-detail.spec.tsx
+++ b/packages/tap-bark/src/components/failure-detail.spec.tsx
@@ -1,0 +1,24 @@
+
+
+import * as React from "react";
+import { configure, shallow } from "enzyme";
+import Adapter from "enzyme-adapter-react-16";
+
+configure({ adapter: new Adapter() });
+
+import { Test, TestFixture, Expect, TestCase } from "../../../alsatian";
+import { FailureDetail } from "./failure-detail";
+
+@TestFixture("<FailureDetails /> Tests")
+export class FailureDetailTests {
+    @TestCase(null)
+    @TestCase(undefined)
+    @TestCase({})
+    @TestCase({ phil: "rob", dawid: "binks" })
+    @Test("weird inputs")
+    public weirdThingsDoesntBreak(value: any) {
+        const result = shallow(<FailureDetail name="something" value={value} />);
+
+        Expect(() => result.html()).not.toThrow();
+    }
+}

--- a/packages/tap-bark/src/components/failure-detail.spec.tsx
+++ b/packages/tap-bark/src/components/failure-detail.spec.tsx
@@ -1,5 +1,3 @@
-
-
 import * as React from "react";
 import { configure, shallow } from "enzyme";
 import Adapter from "enzyme-adapter-react-16";

--- a/packages/tap-bark/src/components/failure-detail.tsx
+++ b/packages/tap-bark/src/components/failure-detail.tsx
@@ -3,12 +3,14 @@ import { Box, Text } from "ink";
 
 export interface FailureDetailProps {
     name: string;
-    value: string;
+    value: any;
 }
 
 export function FailureDetail(props: FailureDetailProps) {
+    const stringValue = JSON.stringify(props.value);
+
     return <>
             <Text underline>{props.name}:</Text>
-            <Box>{props.value}</Box>
+            <Box>{stringValue}</Box>
            </>;
 }

--- a/packages/tap-bark/tsconfig.json
+++ b/packages/tap-bark/tsconfig.json
@@ -7,6 +7,7 @@
     },
     "exclude": [
         "node_modules",
+        "src/**/*.spec.tsx",
         "test/**/*.test.ts",
         "../alsatian"
     ],

